### PR TITLE
fix: node discovery extended, empty host info bug fixed, logging enhanced

### LINKED ISSUE #120: Statistic service does not discover all nodes in the network.
Currently Statistics service query always start with the nodes in the config.  These nodes should only be used for the first query when the DB is empty.  Using the nodes in the DB, this allow the service to discover the entire network

https://github.com/symbol/statistics-service/blob/d78a50c920d8334bbdbb55acc1ed48c4880788fc/src/services/NodeMonitor.ts#L87

The other option is the service will need to loop until it discover all nodes.

### PR DESCRIPTION
**Pull request by yilmazbahadir**
fixes #120 

### What was the issue?
As described in issue #120 Statistics-service was not able to discover and list all the nodes in the network. It was using some static node list in the configuration to query the `node/peers` and those queries were being fired at the same time therefore most of them was timing out.

### What's the fix?
* In addition to the static node list in the configuration(env vars), we are now adding all the nodes in the database to the list and running `node/peers` API calls in this large list.
* Calling `node/peers` endpoint in chunks now, handling the next chunk after the current one is complete, the default chunk size is 50, but it's configurable through the env var `NODE_PEERS_REQUEST_CHUNK_SIZE`
* ChainHeight monitoring task is refactored to process nodes in chunks
* Logging is enhanced
* Known issues => https://github.com/symbol/statistics-service/issues/122

### CODE REVIEW

File: src/config/config.json [hunk 1]
```diff
@@ -17,6 +17,8 @@
 	"PEER_NODE_PORT": 7900,
 	"REQUEST_TIMEOUT": 5000,
 	"NUMBER_OF_NODE_REQUEST_CHUNK": 10,
+	"NODE_PEERS_REQUEST_CHUNK_SIZE": 50,
+	"CHAIN_HEIGHT_REQUEST_CHUNK_SIZE": 10,
```
Comment (AnthonyLaw) [inline src/config/config.json:21]: Do you think can we just create a global chunk size that will work?

File: src/services/NodeMonitor.ts [hunk 2]
```diff
@@ -124,49 +131,81 @@ export class NodeMonitor {
...
+		const apiNodeList = this.nodeList.filter((node) => isAPIRole(node.roles));
 
-		this.nodeList = [];
+		logger.info(
+			`[fetchAndAddNodeListPeers] Getting peers from nodes, total nodes: ${this.nodeList.length}, api nodes: ${apiNodeList.length}`,
+		);
 
-		for (const nodes of nodeListChunks) {
-			logger.info(`Getting node info for chunk of ${nodes.length} nodes`);
+		await runTaskInChunks(this.nodeList, this.nodePeersChunkSize, logger, 'fetchAndAddNodeListPeers', async (nodes) => {
```
Comment (AnthonyLaw) [inline src/services/NodeMonitor.ts:150]: It should use `apiNodeList` instead of `this.nodeList`?

File: src/services/NodeMonitor.ts [hunk 3]
```diff
@@ -81,31 +90,29 @@ export class NodeMonitor {
...
-				return this.fetchNodePeersByURL(hostUrl);
-			}
+		// Fetch node list from database
+		const nodesFromDb = (await DataBase.getNodeList().then((nodes) => nodes.map((n) => n.toJSON()))) || [];
 
-			return [];
-		});
+		logger.info(`[getNodeList] Nodes count from DB: ${nodesFromDb.length}`);
+		// adding the nodes from DB to the node list
+		this.addNodesToList(nodesFromDb);
```
Comment (OlegMakarenko) [inline src/services/NodeMonitor.ts:109]: Seems like dead / disconnected nodes can stay in the list?

Comment (yilmazbahadir) [reply src/services/NodeMonitor.ts:109]: It's a known issue, explained in the PR description, addressed in issue https://github.com/symbol/statistics-service/issues/122

File: src/config/config.json (hunk 1 above)
Comment (yilmazbahadir) [reply src/config/config.json:21]: I specifically avoided that, there wouldn't be just one chunk size that will fit all jobs. Managing it in different variables will allow us to adjust it individually in the runtime(via env vars).

File: src/services/NodeMonitor.ts (hunk 2 above)
Comment (yilmazbahadir) [reply src/services/NodeMonitor.ts]: Interesting, I specifically created the `apiNodeList` for this use, must have been regressed during the refactoring, fixed now.

**Review by OlegMakarenko — APPROVED** [review]

**Comment by AnthonyLaw** [comment]
Look good mate.

but I believe we need this solution asap https://github.com/symbol/statistics-service/issues/122.
It will affect the node count issue here https://github.com/symbol/statistics-service/blob/d78a50c920d8334bbdbb55acc1ed48c4880788fc/src/services/NodeMonitor.ts#L147

File: src/config/config.json (hunk 1 above)
Comment (0x6861746366574) [reply src/config/config.json:21]: Good thinking, @yilmazbahadir.

File: src/config/config.json [hunk 4]
```diff
@@ -17,6 +17,8 @@
...
 	"PREFERRED_NODES": ["*.symboldev.network"],
```
Comment (0x6861746366574) [inline src/config/config.json:22]: Mmm? Is this correct?

Comment (0x6861746366574) [inline src/config/config.json:22]: Is this correct naming?

**Review by 0x6861746366574 — APPROVED** [review]

File: src/config/config.json (hunk 4 above)
Comment (yilmazbahadir) [reply src/config/config.json:22]: As we discussed in the discord, here is the issue created. https://github.com/symbol/statistics-service/issues/125

**Pull request merged**